### DISCUSSION TITLE
Unit testing for undici.fetch

### DISCUSSION THREAD

**Discussion by jennyEckstein** [nodejs/undici#1054] [Q&A] [answered]
I am building Node.js http-client  that uses undici.fetch under the hood. I am looking for a solution like [nock](https://github.com/nock/nock). I see that undici has [MockClient](https://github.com/nodejs/undici/blob/main/docs/api/MockClient.md) for that purpose. Should I use MockClient if I am using undici.fetch, or is it only for non-fetch undici?

So far I tried 'fetch-mock-jest', it seems to do the job but not well. I would love to hear your recommendation about how to test properly.

```javascript
jest.mock('undici', () => ({
  ...jest.requireActual('undici'),
  fetch: jest.fn(() => require('fetch-mock-jest').sandbox())
}));
```

**Comment by mcollina** [comment, marked as answer]
I would recommend to follow https://undici.nodejs.org/#/docs/best-practices/mocking-request. All things apply to fetch as request as they are underpinned by the same logic.
